storops/unity/resource/pool.py

```diff
@@ -203,6 +206,51 @@ def disk_groups(self):
                 dgs[pd.disk_group.get_id()] = [pd]
         return dgs
 
+    @version('<5.1')
+    def is_compression_enabled(self):
+        return self.is_all_flash
+
+    @version('>=5.1')  # noqa
+    def is_compression_enabled(self):
+        resp = self._cli.action(self.resource_class, self.get_id(),
+                                'verifyDataReduction')
+        resp.raise_if_err()
+        if resp.body.get('content') and resp.body.get('content').get(
+                'verifyPass') is True:
+            return True
+        return False
+
+    def verify_advanced_dedup_by_unity_model(self, unity_support_matrix):
+        from storops.unity.resource.system import UnitySystem
+        unity_system = UnitySystem(cli=self._cli)
+        unity_model = unity_system.model.split()[-1]
+        return unity_model in unity_support_matrix and self.is_all_flash
+
+    @version('<4.5')
+    def is_advanced_dedup_enabled(self):
+        return False
+
+    @version('=4.5')  # noqa
+    def is_advanced_dedup_enabled(self):
+        unity_support_matrix = ['450F', '550F', '650F']
```

Comment:
Can we add a decorator for `unity_models` like `version`?
So, we can share the code in other places where we need to check the unity model.

Reply:
Done, added a UnityModel class to provide model compare operations.